src/loaders/tvg/tvgTvgLoadParser.cpp

```diff
@@ -0,0 +1,676 @@
+ * SOFTWARE.
+ */
+
+#include <string.h>
+#include "tvgTvgLoadParser.h"
+
+// Enumeration specifying the loading result
+enum class LoaderResult { InvalidType, Success, SizeCorruption, MemoryCorruption, LogicalCorruption };
+
+// Load paint from .tvg binary file.
+static LoaderResult _parsePaint(tvgBlock block, Paint ** paint);
+
+// Read tvgBlock from pointer location.
+static tvgBlock _readTvgBlock(const char *pointer)
+{
+    tvgBlock block;
+    block.type = *pointer;
+    _read_tvg_ui32(&block.lenght, pointer + TVG_INDICATOR_SIZE);
+    block.data = pointer + TVG_INDICATOR_SIZE + BYTE_COUNTER_SIZE;
+    block.blockEnd = block.data + block.lenght;
+    return block;
+}
+
+/*
+ * Read header of the .tvg binary file
+ * Returns true on success, false otherwise.
+ * Details:
+ * Each file starts with "TVG" phrase and three chars of version number.
+ * Two next bytes are the metadata length (amount of bytes to skip if the meta is not read).
+ * Metadata may contain file generation time and date or other information about device and file.
+ * ["TVG"][version eg. "01a"][uint16 metadata length eg. 0x0000][metadata]
+ *
+ * Starting phase together with metadata are called header.
+ * After header, binary section starts.
+ * NOTE: In whole file little endian is used.
+ */
+static bool _readTvgHeader(const char **pointer)
+{
+    // Sign phase, always "TVG" declared in TVG_HEADER_TVG_SIGN_CODE
+    if (memcmp(*pointer, TVG_HEADER_TVG_SIGN_CODE, TVG_HEADER_TVG_SIGN_CODE_LENGTH)) return false;
+    *pointer += TVG_HEADER_TVG_SIGN_CODE_LENGTH; // move after sing code
+
+    // Standard version number, declared in TVG_HEADER_TVG_VERSION_CODE
+    if (memcmp(*pointer, TVG_HEADER_TVG_VERSION_CODE, TVG_HEADER_TVG_VERSION_CODE_LENGTH)) return false;
+    *pointer += TVG_HEADER_TVG_VERSION_CODE_LENGTH; // move after version code
+
+    // Matadata phase
+    uint16_t meta_lenght; // Matadata phase lenght
+    _read_tvg_ui16(&meta_lenght, *pointer);
+    *pointer += 2; // move after lenght
+
+#ifdef THORVG_LOG_ENABLED
+    char metadata[meta_lenght + 1];
+    memcpy(metadata, *pointer, meta_lenght);
+    metadata[meta_lenght] = '\0';
+    printf("TVG_LOADER: Header is valid, metadata[%d]: %s.\n", meta_lenght, metadata);
+#endif
+
+    *pointer += meta_lenght;
+    return true;
+}
+
+// Paint
+/*
+ * Load paint's cmpMethod and cmpTarget from .tvg binary file.
+ * Returns LoaderResult::Success on success or LoaderResult::SizeCorruption / MemoryCorruption if corrupted.
+ */
+static LoaderResult _parseCmpTarget(const char *pointer, const char *end, Paint *paint)
+{
+    tvgBlock block = _readTvgBlock(pointer);
+    if (block.blockEnd > end) return LoaderResult::SizeCorruption;
+
+    CompositeMethod cmpMethod;
+    if (block.type != TVG_PAINT_CMP_METHOD_INDICATOR) return LoaderResult::LogicalCorruption;
+    if (block.lenght != sizeof(TvgFlag)) return LoaderResult::SizeCorruption;
+    switch (*block.data)
+    {
+        case TVG_PAINT_CMP_METHOD_CLIPPATH_FLAG:
```

Comment:
can't we use CompositeMethod value directly? then we can avoid switch-case

Reply:
We discussed it already with @JSUYA, he also suggested direct direct value passing. However, I believer that for future backward compatibility, data corruption checking and general code readability using switch-case has pros